Bottom: 170820b8c95d6b36a07975fb846ba32cf6d1ca3c
Top:    69415a28efff0c2cde93af2e70b889bab4f010cd
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-09-28 12:31:19 +0800

e1000: use alias for default model

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 461c128..ce68cba 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -1583,7 +1583,7 @@ void pc_nic_init(ISABus *isa_bus, PCIBus *pci_bus)
         if (!pci_bus || (nd->model && strcmp(nd->model, "ne2k_isa") == 0)) {
             pc_init_ne2k_isa(isa_bus, nd);
         } else {
-            pci_nic_init_nofail(nd, pci_bus, "e1000", NULL);
+            pci_nic_init_nofail(nd, pci_bus, "e1000-82540em", NULL);
         }
     }
 }
diff --git a/hw/net/e1000.c b/hw/net/e1000.c
index 09c9e9d..7e87249 100644
--- a/hw/net/e1000.c
+++ b/hw/net/e1000.c
@@ -1666,11 +1666,6 @@ static const E1000Info e1000_devices[] = {
     },
 };
 
-static const TypeInfo e1000_default_info = {
-    .name          = "e1000",
-    .parent        = "e1000-82540em",
-};
-
 static void e1000_register_types(void)
 {
     int i;
@@ -1688,7 +1683,6 @@ static void e1000_register_types(void)
 
         type_register(&type_info);
     }
-    type_register_static(&e1000_default_info);
 }
 
 type_init(e1000_register_types)
diff --git a/hw/pci/pci.c b/hw/pci/pci.c
index b0bf540..a6f3eb2 100644
--- a/hw/pci/pci.c
+++ b/hw/pci/pci.c
@@ -1638,6 +1638,7 @@ static const char * const pci_nic_models[] = {
     "i82559er",
     "rtl8139",
     "e1000",
+    "e1000-82540em",
     "pcnet",
     "virtio",
     NULL
diff --git a/qdev-monitor.c b/qdev-monitor.c
index 0bf7f83..69bce78 100644
--- a/qdev-monitor.c
+++ b/qdev-monitor.c
@@ -50,6 +50,7 @@ static const QDevAlias qdev_alias_table[] = {
     { "lsi53c895a", "lsi" },
     { "ich9-ahci", "ahci" },
     { "kvm-pci-assign", "pci-assign" },
+    { "e1000-82540em", "e1000"},
     { }
 };
